aptos-move/e2e-move-tests/src/tests/governance_updates.rs

```diff
@@ -69,6 +72,77 @@ fn large_transactions() {
     assert!(status.is_discarded());
 }
 
+#[test]
+fn alt_execution_limit_for_gov_proposals() {
+    // This test validates that approved governance scripts automatically gets the
+    // alternate (usually increased) execution limit.
+    let max_gas_regular = 10;
+    let max_gas_gov = 100;
+
+    // Set up the testing environment
+    let mut h = MoveHarness::new();
+
+    let alice = h.new_account_at(AccountAddress::from_hex_literal("0xa11ce").unwrap());
+    let root = h.aptos_framework_account();
+
+    h.modify_gas_schedule(|gas_params| {
+        let txn = &mut gas_params.vm.txn;
+
+        txn.max_execution_gas = Gas::new(10).to_unit_with_params(txn);
+        txn.max_execution_gas_gov = Gas::new(100).to_unit_with_params(txn);
+    });
+    h.set_resource(
+        *root.address(),
+        ApprovedExecutionHashes::struct_tag(),
+        &ApprovedExecutionHashes { entries: vec![] },
+    );
+
+    // Compile the test script, which contains nothing but an infinite loop.
+    let package = BuiltPackage::build(
+        common::test_dir_path("infinite_loop.data/empty_loop_script"),
+        BuildOptions::default(),
+    )
+    .expect("should be able to build package");
+    let script = package
+        .extract_script_code()
+        .pop()
+        .expect("should be able to get script");
+
+    // Execute the script. The amount of gas used should fall within the regular limit.
+    let txn = h.create_script(&alice, script.clone(), vec![], vec![]);
+    let output = h.run_raw(txn);
+    assert_eq!(
+        output.status(),
+        &TransactionStatus::Keep(ExecutionStatus::MiscellaneousError(Some(
+            StatusCode::EXECUTION_LIMIT_REACHED
+        ))),
+    );
+    let gas_used = output.gas_used();
+    assert!(max_gas_regular - 1 <= gas_used && gas_used <= max_gas_regular + 1);
```

Comment:
Good catch. I was originally thinking that there was an off by 1 error caused by rounding, but that was the wrong reason.

The reason why we're having this is because when a transaction gets terminated due to hitting execution gas limit, there could an overshoot, as long as the overall gas budget hasn't been depleted.

I've changed the formula to `let overshoot = (max_gas_regular.min(max_gas_gov) / 5).max(1);`, which should be a bit more robust than a constant value 1.